Add used memory amount to CircuitBreakingException message (#22521)

bumbu opened the pull request:
Closes #22521 

Tests don't check for the amount of used memory because that will tie-up the tests to the implementation. 
From what I saw there are no tests to check that the used memory is calculated correctly, but it may be just my lack of knowledge of the codebase. Also it is out of the scope of original ticket.

dakrone commented:
@bumbu thanks for this! This looks pretty good, could you add a unit test to `MemoryCircuitBreakerTests` with exact numbers so we can assert the message looks correct? (I understand you can't easily add it to the integration tests since the desired memory is not constant)

bumbu commented:
Hi @dakrone,

I added an assertion to `MemoryCircuitBreakerTests` to check for the numbers returned in the message. I didn't hard-code the number, instead I computed it right before the assertion. Didn't want to hard-code the number as how it is currently done gives more insight on what that number is. 

Also I tested only for the first part of the number value (aka testing only for `1024` from `[1024/1k]`) as I didn't want to add one more dependency for computing the byte size value. The assertion should cover the message pretty well as the same number as not formatted and formatted byte size value. But I can easily add that if it will bring more value. 

I squashed the commit.

dakrone commented on core/src/test/java/org/elasticsearch/common/breaker/MemoryCircuitBreakerTests.java:
```diff
@@ -235,6 +235,9 @@ public void testConstantFactor() throws Exception {
             assertThat("breaker was tripped exactly twice", breaker.getTrippedCount(), equalTo(2L));
+
+            long newUsed = (long)(breaker.getUsed() * breaker.getOverhead());
+            assertThat(cbe.getMessage().contains("would be [" + newUsed "/"), equalTo(true));
```
This is missing a `+` between `newUsed` and `"/"`, so it's not compiling currently

bumbu replied:
My bad. I was sure that it's not possible to introduce bugs with such a small change... Lesson learned.

bumbu commented:
Hi @dakrone.
Sorry for the delay, I was on holidays. 
I fixed the bug, run tests, squashed commits and rebased from master. 
Please check.

dakrone commented:
@elasticmachine test this please

dakrone commented:
Thanks @bumbu, I've merged this!
